### PR TITLE
Set the ufrag/pwd according to the spec

### PR DESCRIPTION
**Pull request by tomaka**
Seeing the small discussion in the WebRTC implementation PR, I realized that smoldot doesn't actually say what is in the spec now. This PR changes it.

### CODE REVIEW

**Comment by tomaka** [comment]
Doesn't work on Chrome:

> Failed to execute 'setLocalDescription' on 'RTCPeerConnection': Failed to set local offer sdp: Failed to apply the description for m= section with mid='0': Invalid ICE parameters: ICE ufrag must contain only alphanumeric characters, '+', and '/'.

File: bin/wasm-node/javascript/src/index-browser.ts [hunk 1]
```diff
@@ -315,10 +319,9 @@ export function start(options?: ClientOptions): Client {
               "a=ice-options:ice2" + "\n" +
               // ICE username and password, which are used for establishing and
               // maintaining the ICE connection. (RFC8839)
-              // MUST match ones used by the answerer (server).
               // These values are set according to the libp2p WebRTC specification.
-              "a=ice-ufrag:" + remoteCertMultibase + "\n" +
-              "a=ice-pwd:" + remoteCertMultibase + "\n" +
+              "a=ice-ufrag:" + ufragPwd + "\n" +
```
Comment (melekes) [inline bin/wasm-node/javascript/src/index-browser.ts:323]: I'm somewhat disagree with this

**Review by melekes — APPROVED** [review]

**Comment by tomaka** [comment]
Since Chrome throws an exception, it's clear that it's going to be changed again, and I'm not merging this PR beforehand.

File: bin/wasm-node/javascript/src/index-browser.ts [hunk 2]
```diff
@@ -267,13 +267,17 @@ export function start(options?: ClientOptions): Client {
...
-          if (pwd != null) {
-            sdpOffer = sdpOffer.replace(/^a=ice-ufrag.*$/m, 'a=ice-ufrag:' + pwd[1]);
-          } else {
+          // randomly-generated string on both sides, and must be prefixed with
+          // `libp2p-webrtc-v1:`. We modify the local description to ensure that.
+          // While we could randomly generate a new string, we just grab the one that the
+          // browser has generated, in order to make sure that it respects the constraints
+          // of the ICE protocol.
+          const ufragPwd = "libp2p-webrtc-v1:" + sdpOffer.match(/^a=ice-pwd:(.+)$/m)?.at(1);
+          if (ufragPwd === undefined) {
```
Comment (tomaka) [inline bin/wasm-node/javascript/src/index-browser.ts:276]: Note for myself: since we do `"libp2p-webrtc-v1:" + ...`, this can never actually be undefined

**Comment by tomaka** [comment]
I'm going to merge this because I'm releasing a new version.
We can always open another PR if the spec changes again.

**Pull request merged**